tests/racket/conditions005/Main.idr

```diff
@@ -0,0 +1,22 @@
+-- Idris2
+
+import System
+import System.Concurrency
+
+-- Test `conditionBroadcast` wakes all threads for 1 main and N child threads
+
+main : IO ()
+main =
+  let n = 3 in
+  do cvMutex <- makeMutex
+     cv <- makeCondition
+     ts <- for [1..n] $ \_ => fork $ do mutexAcquire cvMutex
+                                        conditionWait cv cvMutex
+                                        putStrLn "Hello mother"
+                                        mutexRelease cvMutex
+     putStrLn "Hello children"
+     sleep 1
+     conditionBroadcast cv
+     for ts $ \t => threadWait t
```

Comment:
`IO` actions are not allowed to silently discard their result
anymore. So you need an `ignore $` in front of this `for (...)`.

Reply:
I thought I'd fixed that, but I must have undone it accidentally. It should be fixed now.

I re-enabled and ran the `Futures` tests, and whatever the issue was, it seems to be gone now. Hopefully it is safe to merge now, but let me know what you think  : )

Reply:
Great news! By the way, I checked and we do have

```idris
for_ : (Foldable t, Applicative f) => t a -> (a -> f b) -> f ()
```

but it does not matter whether we use one or the other as they're exactly the same AFAICR.